src/mercury_engine_data_structures/version_validation.py

```diff
@@ -0,0 +1,62 @@
+from hashlib import md5
+
+from construct.core import BytesInteger, Int32ul, PrefixedArray, Struct
+
+from mercury_engine_data_structures.file_tree_editor import FileTreeEditor, GameVersion
+from mercury_engine_data_structures.formats.property_enum import FileNameEnum
+from mercury_engine_data_structures.formats.toc import Toc
+
+MANIFEST_CONSTRUCT = Struct(
+    "toc_md5" / BytesInteger(16),
+    "file_data" / PrefixedArray(Int32ul, Struct("assetid" / FileNameEnum, "md5" / BytesInteger(16))),
+)
+
+
+def get_md5(data: bytes) -> int:
+    return int(md5(data, usedforsecurity=False).hexdigest(), 16)
+
+
+def get_expected_assetids(editor: FileTreeEditor) -> dict[int, str]:
+    manifest = MANIFEST_CONSTRUCT.parse_file(editor.version.manifest, target_game=editor.target_game)
+
+    return {editor.target_game.hash_asset(file.assetid): file.assetid for file in manifest.file_data}
+
+
+def check_version(editor: FileTreeEditor) -> tuple[bool, GameVersion]:
+    toc_hash = get_md5(editor.root.joinpath(Toc.system_files_name()).read_bytes())
+    for ver in GameVersion:
+        if ver.toc_hash == toc_hash:
+            print(f"VERSION {ver}")
+            return True, ver
+
+    return False, GameVersion.UNIDENTIFIED
```

Comment:
Would just raise an exception when it's an unknown version. Would lead to not needing the True/False result.

(Could also have done with unidentified being None instead)